internal/etag/reader.go

```diff
@@ -153,3 +161,42 @@ type VerifyError struct {
 func (v VerifyError) Error() string {
 	return fmt.Sprintf("etag: expected ETag %q does not match computed ETag %q", v.Expected, v.Computed)
 }
+
+// UUIDHash - use uuid to make md5sum
+type UUIDHash struct {
+	uuid []byte
+}
+
+// Write -  implement hash.Hash Write
+func (u UUIDHash) Write(p []byte) (n int, err error) {
+	return len(p), nil
+}
+
+// Sum -  implement md5.Sum
+func (u UUIDHash) Sum(b []byte) []byte {
+	hasher := md5.New()
+	hasher.Write(u.uuid)
+	return hasher.Sum(nil)
+}
```

Comment:
```suggestion
func (u UUIDHash) Sum(b []byte) []byte {
	return u.uuid
}
```

You are hashing the uuid - not needed.